src/main/java/com/onlinebookshop/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.onlinebookshop.repository.impl;
+
+import com.onlinebookshop.exception.DataProcessingException;
+import com.onlinebookshop.model.Book;
+import com.onlinebookshop.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
Annotation is not necessary here.  Spring will understand it anyway.

Reply:
Yes, thank you for review, i know that Spring will automatically autowire construcrute if we have only 1 in our Entity. But in my opinion it`s more clearly, explicitly when we`re using annotation.